### PR TITLE
Set up opam install

### PR DESCRIPTION
**Pull request by ihodes**

### CODE REVIEW

File: opam/files/hlarp.install [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bin: [
+     "hlarp"
```
Comment (rleonid) [inline opam/files/hlarp.install]: shouldn't this be `hlarp_cli.native` ?

Comment (rleonid) [inline opam/files/hlarp.install]: Nevermind, just looked at `build`

Comment (rleonid) [inline opam/files/hlarp.install]: Nevermind, just looked at `build` target

File: opam/opam [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+version: "0.0.0"
+maintainer: "Leonid Rozenberg <leonidr@gmail.com>"
+authors: ["Isaac Hodes <isaachodes@gmail.com>"
+          "Leonid Rozenberg <leonidr@gmail.com>"]
+homepage: "https://github.com/hammerlab/hlarp"
+bug-reports: "https://github.com/hammerlab/hlarp/issues"
+license: "Apache-2.0"
+build: [
+  [make]
+]
```
Comment (rleonid) [inline opam/opam]: You don't need to nest the `make` command here. `build: [make]` should work.

File: Makefile [hunk 3]
```diff
@@ -4,24 +4,33 @@ PACKAGES_INSTALL=$(PACKAGES)
...
+	    _build/src/lib/hlarp.a\
+	    _build/src/lib/hlarp.o\
+	    _build/src/lib/hlarp.cma\
+	    _build/src/lib/hlarp.cmi\
+	    _build/src/lib/hlarp.cmo\
+	    _build/src/lib/hlarp.cmx\
+	    _build/src/lib/hlarp.cmxa\
+            _build/src/lib/hlarp.cmxs
+uninstall:
+	ocamlfind remove ogene
```
Comment (rleonid) [inline Makefile]: you mean `remove hlarp` :)

File: opam/opam [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+install: [make "install"]
+remove: [make "uninstall"]
+depends: [
+  "ocamlfind"
+  "cmdliner"
+  "nonstd"
+  "re"
+  "re.posix"
+  "re.glob"
```
Comment (rleonid) [inline opam/opam]: 1. I don't think that `re.posix` and `re.glob` are independent opam packages, `re` should be fine.
2. `ocamlfind` can have a `{build}` qualifier on it.
3. We also depend on `ocamlbuild` which also should have a `{build}` qualifier on it.

File: Makefile [hunk 5]
```diff
@@ -4,24 +4,33 @@ PACKAGES_INSTALL=$(PACKAGES)
 
 SOURCE_DIRS=/util /unc /stats /cls /rgr /uns
```
Comment (rleonid) [inline Makefile]: Can you please also remove this line? It is a carry over of when I copied the `Makefile` from `oml`. So my bad.

**Comment by ihodes** [comment]
Thanks for the review; I didn't know what {build} did: that's a good one to know. 

I think I covered everything mentioned.

File: Makefile [hunk 6]
```diff
@@ -1,27 +1,33 @@
-
 PACKAGES=nonstd re re.posix re.glob cmdliner
 PACKAGES_INSTALL=$(PACKAGES)
```
Comment (rleonid) [inline Makefile]: I think that you forgot to remove (or filter out) `re.posix` and `re.glob` here ?

Comment (ihodes) [inline Makefile]: Fixed.

**Pull request merged**